config/config-api/src/com/thoughtworks/go/domain/UnitTestReportGenerator.java

```diff
@@ -154,8 +156,8 @@ private void pumpFileContent(File file, PrintStream out) throws IOException {
         BufferedReader bufferedReader = null;
         try {
             bufferedReader = new BufferedReader(new FileReader(file));
-            String line = bufferedReader.readLine();
-            if (!line.contains("<?xml")) { // skip prolog
+            String line = PROLOG_PATTERN.matcher(bufferedReader.readLine()).replaceFirst("");
+            if (!line.isEmpty()) { // skip if line was only prolog
```

Comment:
I think your change looks good! Just one nitpick, perhaps it's my discomfort with not knowing if the Regex implementation in java are greedy in nature. Also I'm not sure what'd happen if the prolog string happened to be inside a `CDATA` section (remember `CDATA` does not do escaping)

Could we instead change this to something like (may not compile, but gives you a general idea) –

```java
// string that begins with the prolog, ignoring any whitespace
if (line.matches(/^\s*<\?xml\b/)){
  int prologStartIndex = line.indexOf("<?xml ");
  int prologEndIndex = line.indexOf("?>");
  // not sure if we need to handle a `-1` return value for any of the above?
  if (endIndex > prologStartIndex) {
    prologString = line.substring(prologStartIndex, prologEndIndex);
    line = line.replace(prologString, "");
  }
}
```

WDYT?